src/hotspot/share/runtime/handshake.cpp

```diff
@@ -565,39 +567,31 @@ HandshakeState::ProcessResult HandshakeState::try_process(HandshakeOperation* ma
 
   Thread* current_thread = Thread::current();
 
-  HandshakeState::ProcessResult pr_ret = HandshakeState::_processed;
-  int executed = 0;
+  HandshakeOperation* op = get_op();
```

Comment:
Again why not pop here?

Reply:
See above.

If there was only one OP in queue, the handshake may now disarm it's poll and continue.
If there is more in queue it will still go to slow path.

Reply:
So comment above applies here, we must not let queue become empty until OP is completed.